src/api/getAmmFundingOrderData.js

```diff
@@ -6,7 +6,7 @@ const validateWithJoi = require('../lib/validators/validateWithJoi')
 const schema = Joi.object({
   pool: Joi.string(),
   token: Joi.string(),
-  amount: Joi.bigNumber().greaterThan(0)
+  amount: Joi.string().example('0.1')
```

Comment:
I think it would become the only endpoint where `amount` is `Joi.string()` whereas other endpoints use `Joi.bigNumber()` or `Joi.amount()`. Are you sure it doesn't fit one of the two ?

Reply:
I believe `Joi.bigNumber().greaterThan(0)` is the correct type. User should be able to provide a string though (which would be converted into BigNumber and validated).

Reply:
`Joi.amount` is only applicable to order amounts since some orders are forwarded to bitfinex which limits precision of order amounts to 8 decimal places

Reply:
I don't argue about the proper Joi type to use, but based on testing, using `bigNumber` wasn't working. The two types that worked, were `string` and `number`. I haven't tried with `amount`.
Using `bigNumber` was returning `400 'amount' is required` error from the API.

Reply:
Indeed there seems to be an issue with serialization of `BigNumber`, this might be with the way we do the requests (if we're still using `request-promise` that could be the reason)

Reply:
however looking at other examples using `BigNumber` maybe there was an issue with the way we were sending the request, are all other endpoints deserialized as `BigNumber` as well on the api side?

Reply:
Forgot to mention that the reason this is not converting automatically is because amount here is part of querystring on a GET request which must be handled by `request-promise` differently then json body

Reply:
That could be the main culprit, nice use of [`over`](https://ramdajs.com/docs/#over), I hadn't used that one before
is it possible to make `Joi` do the conversion by setting the type to `string` or will it throw?

Reply:
Updated to convert `amount` to string, after validation and before sending the request. 👌